src/math/fft/fft.c

```diff
@@ -178,3 +178,184 @@ void fft_execute(struct fft_plan *plan, bool ifft)
 			icomplex_shift(&plan->outb[i], plan->len, &plan->outb[i]);
 	}
 }
+
+#ifdef UNIT_TEST
```

Comment:
Should these UT's be in the UT directory ?

Reply:
this seems like a lot of generic code to hide behind ut

Reply:
yes, like @cujomalainey commented, I put it here as a example code about how to use the library with comp_buffer buffers (looks needed for unit test at the moment). They can be removed in someday that proved no needed anymore for either unit test or the real user from other audio components.

Reply:
Hmm, I would at least move to an examples folder or at least an adjacent file that has a DO NOT COMPILE  comment.

Reply:
> Hmm, I would at least move to an examples folder or at least an adjacent file that has a DO NOT COMPILE comment.

Fair enough, let me move this to the UT part.